src/Sarif/EnumeratedArtifact.cs

```diff
@@ -164,7 +166,14 @@ private void TryRewindStream()
                 }
                 else if (Uri != null && Uri.IsAbsoluteUri && Uri.IsFile)
                 {
-                    this.sizeInBytes = (long)FileSystem.FileInfoLength(Uri.LocalPath);
+                    if (FileSystem.IsSymbolicLink(Uri.LocalPath))
+                    {
+                        GetArtifactData();
```

Comment:
>GetArtifactData [](http://example.com/codeflow?start=24&length=15)

So you've broken a core (undocumented!) assumption in this code that we do *not* load the scan target to determine its file size. 

The reason is that the file enumeration phase is single-threaded and it is helpful to defer file loading until the multithreaded scanning phase.

What you need to do here is obtain the actual path pointed to by the symbolic link. And then perform the size in bytes against that location using `FileInfoLength`. 

Speaking of which, why doesn't new FileInfo(path).Length work here? i.e., why would file open read work properly but the length check fails?

Reply:
Fixed as proposed.